File: filesystem/src/main/java/com/nytimes/android/external/fs/filesystem/FileSystemFactory.java
package com.nytimes.android.external.fs.filesystem;

import javax.annotation.Nonnull;

import java.io.File;
import java.io.IOException;

/**
 * Factory for {@link FileSystem}.
 */
public final class FileSystemFactory {
    private FileSystemFactory() {
    }

    /**
     * Creates new instance of {@link FileSystemImpl}.
     *
     * @param root root directory.
     * @return new instance of {@link FileSystemImpl}.
     * @throws IOException
     */
    @Nonnull
    public static FileSystem create(@Nonnull File root) throws IOException {
        return new FileSystemImpl(root);
    }
}

File: filesystem/src/main/java/com/nytimes/android/external/fs3/filesystem/FileSystemFactory.kt
package com.nytimes.android.external.fs3.filesystem

import java.io.File
import java.io.IOException

/**
 * Factory for [FileSystem].
 */
object FileSystemFactory {

    /**
     * Creates new instance of [FileSystemImpl].
     *
     * @param root root directory.
     * @return new instance of [FileSystemImpl].
     * @throws IOException
     */
    @Throws(IOException::class)
    fun create(root: File): FileSystem = FileSystemImpl(root)
}
